code/game/objects/items/lipstick.dm

```diff
@@ -0,0 +1,188 @@
+	name = "jade lipstick"
+	lipstick_color = "#3EB776"
+
+/obj/item/lipstick/black
+	name = "black lipstick"
+	lipstick_color = "#56352F"
+
+/obj/item/lipstick/amberred
+	name = "amberred lipstick"
+	lipstick_color = "#BA2E2A"
+
+/obj/item/lipstick/cherry
+	name = "cherry lipstick"
+	lipstick_color = "#BD1E35"
+
+/obj/item/lipstick/orange
+	name = "orange lipstick"
+	lipstick_color = "#F75F51"
+
+/obj/item/lipstick/gold
+	name = "gold lipstick"
+	lipstick_color = "#DA8118"
+
+/obj/item/lipstick/deepred
+	name = "deepred lipstick"
+	lipstick_color = "#850A1C"
+
+/obj/item/lipstick/pink
+	name = "pink lipstick"
+	lipstick_color = "#E84272"
+
+/obj/item/lipstick/rosepink
+	name = "rosepink lipstick"
+	lipstick_color = "#E2A4B1"
+
+/obj/item/lipstick/nude
+	name = "nude lipstick"
+	lipstick_color = "#E7A097"
+
+/obj/item/lipstick/wine
+	name = "wine lipstick"
+	lipstick_color = "#D25674"
+
+/obj/item/lipstick/peach
+	name = "peach lipstick"
+	lipstick_color = "#D05049"
+
+/obj/item/lipstick/forestgreen
+	name = "forestgreen lipstick"
+	lipstick_color = "#82B33B"
+
+/obj/item/lipstick/skyblue
+	name = "skyblue lipstick"
+	lipstick_color = "#60C2C5"
+
+/obj/item/lipstick/teal
+	name = "teal lipstick"
+	lipstick_color = "#0A857C"
+
+/obj/item/lipstick/custom
+	name = "lipstick"
+
+/obj/item/lipstick/random
+	name = "lipstick"
+
+/obj/item/lipstick/random/Initialize()
+	. = ..()
+	var/list/lipstick_subtypes = subtypesof(/obj/item/lipstick) - /obj/item/lipstick/random
+	var/obj/item/lipstick/chosen_lipstick = pick(lipstick_subtypes)
+	name = initial(chosen_lipstick.name)
+	lipstick_color = initial(chosen_lipstick.lipstick_color)
+
+
+/obj/item/lipstick/attack_self(mob/user as mob)
+	to_chat(user, SPAN_NOTICE("You twist \the [src] [open ? "closed" : "open"]."))
+	open = !open
+	update_icon()
+
+/obj/item/lipstick/attack(mob/victim, mob/user)
+	if(!open)
+		return
+
+	if(!ismob(victim))
+		return
+
+	if(ishuman(victim))
+		var/mob/living/carbon/human/human = victim
+
+		// if they already have lipstick on
+		if(human.lipstick_data)
+			to_chat(user, SPAN_NOTICE("You need to wipe off the old lipstick first!"))
+			return
+
+		var/datum/lipstick_data/lipstick_data = new /datum/lipstick_data(lipstick_color, lipstick_variant)
+		if(human == user)
+			user.visible_message(SPAN_NOTICE("[user] does their lips with \the [src]."), SPAN_NOTICE("You take a moment to apply \the [src]. Perfect!"))
+			human.lipstick_data = lipstick_data
+			human.update_body()
+		else
+			user.visible_message(SPAN_WARNING("[user] begins to do [human]'s lips with \the [src]."), SPAN_NOTICE("You begin to apply \the [src]."))
+			if(do_after(user, 4 SECONDS, human, do_flags = DO_DEFAULT & ~DO_SHOW_PROGRESS & ~DO_BOTH_CAN_TURN))
+				user.visible_message(SPAN_NOTICE("[user] does [human]'s lips with \the [src]."), SPAN_NOTICE("You apply \the [src]."))
+				human.lipstick_data = lipstick_data
+				human.update_body()
+	else
+		to_chat(user, SPAN_NOTICE("Where are the lips on that?"))
```

Comment:
Make this an early return, instead of ismob just use !ishuman, chat message and early return, and then the rest